core/src/main/java/hudson/cli/OnlineNodeCommand.java

```diff
@@ -40,29 +45,61 @@
 @Extension
 public class OnlineNodeCommand extends CLICommand {
 
-    @Argument(metaVar="NAME", usage="Agent name, or empty string for master")
-    public String computerName;
+    @Argument(metaVar = "NAME", usage = "Agent name, or empty string for master", required = true, multiValued = true)
+    private List<String> nodes;
 
     @Override
     public String getShortDescription() {
-
         return Messages.OnlineNodeCommand_ShortDescription();
     }
 
     @Override
     protected int run() throws Exception {
-
+        boolean errorOccurred = false;
         final Jenkins jenkins = Jenkins.getActiveInstance();
 
-        Computer computer = jenkins.getComputer(computerName);
+        final HashSet<String> hs = new HashSet<String>();
+        hs.addAll(nodes);
+
+        List<String> names = null;
+
+        for (String node_s : hs) {
+            Computer computer = null;
+
+            try {
+                computer = jenkins.getComputer(node_s);
 
-        if (computer == null) {
-            throw new IllegalArgumentException(hudson.model.Messages.Computer_NoSuchSlaveExists(computerName, null));
-        } else {
-            computer.cliOnline();
+                if (computer == null) {
+                    if (names == null) {
+                        names = new ArrayList<String>();
```

Comment:
Probably I realized better way how to understand your comment:).

You could mean to place e.g. final List<String> names = new ArrayList<String>(); at the top of method.

Well it is possible, but how to distinguish between these states:
1) computer names aren't calculated yet (now: names == null)
2) we don't find any computer (now: names.size() == 0)

The null is used now as a signal - "hey we need to calculate the collection". If I initialize names at the declaration, I have only option names.size() == 0... or yet another (boolean) variable.

Am I missed sth.?